### PR TITLE
Improve time taken to run the examples by optimizing `vtadmin` build

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR optimizes how `vtadmin` is built and shipped providing a `50%` time improvement when [running the local examples from a release artefact](https://vitess.io/docs/17.0/get-started/local/) and also from a local `git clone`. Running `101_initial_cluster.sh` used to take `76 seconds` on my Mac whereas now it requires `38 seconds`.

As of now, the static file directory (`build`) and the `node_modules` directory are shipped in the release artefact, leading to no dependency download time and no build time required by the end user. Regarding the local examples, the time required to run them is also improved - the `build` and `node_modules` directory are created during `make build`.

### Make build

The `make build` command now takes slightly longer to run as it builds vtadmin in addition to the go binaries. The vtadmin build can be skipped if we set `NOVTADMINBUILD`.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/11799
- Fixes https://github.com/vitessio/vitess/issues/11539

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

File: web/vtadmin/package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "build": "vite build",
     "test": "vitest",
     "preview": "vite preview",
+    "package": "pkg vite.config.ts --output myapp",
```
Comment (frouioui) [inline web/vtadmin/package.json:42]: let's remove this

File: .github/workflows/create_release.yml [hunk 2]
```diff
@@ -22,6 +22,11 @@ jobs:
       with:
         go-version: 1.20.5
 
+    - name: Setup node
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16'
```
Comment (frouioui) [inline .github/workflows/create_release.yml:28]: Needs to be on version 18

File: web/vtadmin/build.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+source "$NVM_DIR/nvm.sh"
+
+output "\nConfiguring Node.js $NODE_VERSION\n"
+nvm install "$NODE_VERSION" || fail "Could not install and use nvm $NODE_VERSION."
+
+npm --prefix "$web_dir" --silent install
+
+VITE_VTADMIN_API_ADDRESS="http://${hostname}:${vtadmin_api_port}" \
+  VITE_ENABLE_EXPERIMENTAL_TABLET_DEBUG_VARS="true" \
+  npm run --prefix "$web_dir" build
```
Comment (ajm188) [inline web/vtadmin/build.sh:52]: no trailing newline here :)

**Review by ajm188 — APPROVED** [review]

File: web/vtadmin/build.sh (hunk 3 above)
Comment (frouioui) [reply web/vtadmin/build.sh]: Fixed in f21b1d0d77, thanks!

**Review by ajm188 — APPROVED** [review]

**Review by deepthi — COMMENTED** [review]
Nice work! Can you also update the `Build from Source` section of the website docs showing how to skip the vtadmin build, and link the relevant PR into this one?

**Comment by frouioui** [comment]
Done @deepthi, I have updated https://github.com/vitessio/website/pull/1511.

File: web/vtadmin/build.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+# Copyright 2023 The Vitess Authors.
```
Comment (mattlord) [inline web/vtadmin/build.sh:2]: Not a big deal obviously, but an empty line here would be easier to read.

**Review by mattlord — COMMENTED** [review]
My only concern is that before this change, you did not need to build Vitess -- and have a golang dev environment and dependencies installed -- in order to run the examples and could instead use the binary packages. Needing to get a vitess dev environment setup and working is a relatively big hurdle for new users that want to play with the examples IMO.

Am I missing or misunderstanding something?

**Review by mattlord — APPROVED** [review]
Thanks for working on this!

**Pull request merged**